### PR TITLE
Issue #3602: Add EngineView.release() for manually releasing session from view.

### PR DESCRIPTION
**Pull request by pocmo**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt [hunk 1]
```diff
@@ -85,10 +85,19 @@ class GeckoEngineView @JvmOverloads constructor(
...
+        currentSession = null
+
+        currentGeckoView.releaseSession()
+    }
+
     override fun onDetachedFromWindow() {
         super.onDetachedFromWindow()
 
-        currentSession?.apply { unregister(observer) }
+        release()
```
Comment (Amejia481) [inline components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt]: I wonder why this change is on nightly but not beta 🤔 ? As beta and nightly have the same behavior for for the`release` function.

Comment (pocmo) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt]: Oh, yeah, that's something I need to port over to beta.

Comment (pocmo) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt]: Updated!
